src/tools/pub/dependencies.md

```diff
@@ -114,15 +113,31 @@ This example specifies that your package depends on a hosted package named
 `transmogrify` and will work with any version from 1.4.0 to 2.0.0
 (but not 2.0.0 itself).
 
-If you want to use your own package server, you can use a description that
+If you want to use your own package repository, you can use a description that
 specifies its URL:
 
 {% prettify yaml tag=pre+code %}
+# This syntax requires language version 2.15 or higher
+environment: 
+  sdk: >=2.15.0 < 3.0.0
+
+dependencies:
+  transmogrify:
+    hosted: http://some-package-server.com
+    version: ^1.4.0
+{% endprettify %}
+
+Before Dart language version 2.15, the format is different:
```

Comment:
Depending on when this PR is merged, do we want to separate the old version with a version-note mentioning language version like we do in https://github.com/dart-lang/site-www/blob/master/src/_guides/language/analysis-options.md?plain=1#L195-L201? To be consistent with elsewhere on the website. Then in the currently supported version above, don't add the "# This syntax requires language version 2.15 or higher" comment?

Reply:
Done